AIROCFILE-35 - CI: Add hipFile IO test using aiscp

kurtmcmillan opened the pull request:
Test hipFile IO using aiscp to ensure IO works in some capacity. Future changes will allow testing with specific backends.

kurtmcmillan committed:
CI: Test hipFile IO using aiscp

Future changes will allow testing with a specific backend.

riley-dixon commented on util/ci-aiscp-test.sh:
```diff
@@ -0,0 +1,42 @@
+
+set -e
+
+trap 'if [ -e "$TMPDIR" ]; then rm -fr "$TMPDIR"; fi' EXIT
```
I would consider adding a cleanup step in the CI YAML with the `always()` condition. I do not know if this TRAP is guaranteed to execute in all cases (e.g. if GitHub aborts this job mid-step).

While doing so, I would just be mindful as well that the directory created here will be root owned.

kurtmcmillan replied:
Thank you for the suggestions.

- I've updated the github workflow to control the lifetime of the directory used for IO tests
- IO tests are launched from within the IO test directory

riley-dixon replied:
Looks good, thanks!

kurtmcmillan committed:
Documentation fix

kurtmcmillan committed:
Update script variable to match documentation

kurtmcmillan committed:
review: Address PR comments

- The github job now manages the lifetime of the IO test directory which
  ensures it is always cleaned up (even when the job is cancelled)
- IO tests still manage their own temporary directory within the IO test
  directory to ensure that disk space is released at the end of the test.

kurtmcmillan committed:
Add missing container name to docker commands in build-ais.yml

kurtmcmillan committed:
Use double quotes to allow variable substitution

kurtmcmillan committed:
Fix path to ci-aiscp-test.sh
